yaml/src/main/java/com/github/olestxcode/flyconf/loader/YamlConfigurationLoader.java

```diff
@@ -1,52 +1,26 @@
 package com.github.olestxcode.flyconf.loader;
 
 import com.github.olestxcode.flyconf.exception.InvalidConfigurationException;
+import com.github.olestxcode.flyconf.io.InputStreamSource;
+import org.jetbrains.annotations.NotNull;
 import org.yaml.snakeyaml.Yaml;
 
-import java.io.*;
-import java.util.*;
-import java.util.function.Supplier;
-import java.util.stream.Collectors;
+import java.io.InputStream;
+import java.util.Map;
 import java.util.stream.Stream;
 
-public class YamlConfigurationLoader implements PropertyMapLoader {
+public final class YamlConfigurationLoader extends AbstractPropertyMapLoader {
 
-    private final Supplier<InputStream> inputStreamSupplier;
-
-    public YamlConfigurationLoader(File file) {
-        this.inputStreamSupplier = () -> {
-            try {
-                return new FileInputStream(file);
-            } catch (FileNotFoundException e) {
-                throw new RuntimeException(e);
-            }
-        };
-    }
-
-    public YamlConfigurationLoader(Supplier<InputStream> inputStreamSupplier) {
-        this.inputStreamSupplier = inputStreamSupplier;
+    public YamlConfigurationLoader(@NotNull InputStreamSource streamSource) {
+        super(streamSource);
     }
 
     @Override
-    public Map<String, Object> load() {
+    protected Stream<Map.Entry<String, Object>> load(InputStream stream) {
```

Comment:
Won't this level of access affect on method usage outside?

Reply:
No